unsafe-merge-queue not rewriting oops with "Originally-landed-as"

briannafan opened the pull request:
#### cc1b3c8bf6c25c8852f907df58125caa13e5f7b2
<pre>
unsafe-merge-queue not rewriting oops with &quot;Originally-landed-as&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=283086">https://bugs.webkit.org/show_bug.cgi?id=283086</a>
<a href="https://rdar.apple.com/139833841">rdar://139833841</a>

Reviewed by Aakash Jain.

Do not hide AddReviewerToCommitMessage when it is skipped.
Add a reason as to why this step was skipped if it is.

* Tools/CISupport/ews-build/steps.py:
(AddReviewerToCommitMessage.__init__):
(AddReviewerToCommitMessage.run):
(AddReviewerToCommitMessage.getResultSummary):
(AddReviewerToCommitMessage.doStepIf): Deleted.
(AddReviewerToCommitMessage.hideStepIf): Deleted.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/290215@main">https://commits.webkit.org/290215@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/966aa9a29963a88591a66d09ebfa8ddd3cdc1fa5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89329 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8854 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/44168 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/94315 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/40090 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/91380 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/130/builds/9241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/17099 "Failed to checkout and rebase branch from PR 39633") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/94315 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/26488 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92331 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/130/builds/9241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/81083 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/94315 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/130/builds/9241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/35468 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39197 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/130/builds/9241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/36453 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/96144 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/16509 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/123/builds/17099 "Failed to checkout and rebase branch from PR 39633") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/96144 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/88915 "Passed tests") | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/16765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/76874 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/96144 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/20027 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9659 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13998 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/16523 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/16264 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/19715 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/18045 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -6587,10 +6587,18 @@ class AddReviewerToCommitMessage(shell.ShellCommand, AddReviewerMixin):
+        classification = self.getProperty('classification', [])
+        if not isinstance(classification, list):
+            classification = []
+        if not self.getProperty('valid_reviewers') or ['Cherry-pick'] == classification:
```
These two nested if statements can be converted to sequential if statements. That would make it more clear.

e.g.:
```
if ['Cherry-pick'] == classification:
    self.skip_reason = 'commit is a cherry-pick'
    defer.returnValue(SKIPPED)
if not self.getProperty('valid_reviewers'):
    self.skip_reason = 'there are no valid reviewers'
    defer.returnValue(SKIPPED)
```

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -6613,19 +6621,12 @@ def run(self, BufferLogObserverClass=logobserver.BufferLogObserver):
         if self.results == FAILURE:
             return {'step': 'Failed to apply reviewers'}
+        if self.results == SKIPPED:
+            return {'step': f'Skipped applying reviewers because {self.skip_reason}'}
```
can make this message compact (to improve readability in the buildbot build page), e.g.: f'Skipped as {self.skip_reason}'

aj062 reviewed (changes requested):
This is a good step to help us with debugging. However, this doesn't seem enough for our users. They will see the confusing error message: "Commit message contains (OOPS!)". We are not fixing the problem, just make it easier to debug. Can we automatically fix the problem as well?

If not, we should atleast update the error message in `ValidateCommitMessage::run()` so that it also indicate the reason for failure (e.g.: cherry-pick) and maybe suggest what they can do next to fix the issue.

briannafan committed:
unsafe-merge-queue not rewriting oops with "Originally-landed-as"
https://bugs.webkit.org/show_bug.cgi?id=283086
rdar://139833841

Reviewed by Aakash Jain.

Do not hide AddReviewerToCommitMessage when it is skipped.
Add a reason as to why this step was skipped if it is.

* Tools/CISupport/ews-build/steps.py:
(AddReviewerToCommitMessage.__init__):
(AddReviewerToCommitMessage.run):
(AddReviewerToCommitMessage.getResultSummary):
(AddReviewerToCommitMessage.doStepIf): Deleted.
(AddReviewerToCommitMessage.hideStepIf): Deleted.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: https://commits.webkit.org/290215@main
